test/inductor/test_cpu_select_algorithm.py

```diff
@@ -829,6 +840,245 @@ def forward(self, x):
         else:
             self._check_amx_counter(vec_amx)
 
+    @requires_vectorization
+    @unittest.skipIf(
+        "asimd" in [str(vec_isa) for vec_isa in cpu_vec_isa.valid_vec_isa_list()],
+        "Does not support on NEON or SVE",
+    )
+    @dtypes(torch.bfloat16, torch.half, torch.float, torch.int8)
+    def test_microgemm_fp32_vec(self, dtype):
```

Comment:
I feel this test case is a bit hacky, as it exposes too many implementation details.